src/main/java/seedu/address/model/person/contact/Telegram.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.model.person.contact;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Person's telegram in the contact book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTelegram(String)}
+ */
+public class Telegram extends Contact {
+
+    public static final String MESSAGE_CONSTRAINTS = "Telegram username should be of the format @username "
+        + "and adhere to the following constraints:\n"
+        + "1. The username should be between 5 and 32 characters in length."
+        + "2. The username should only contain alphanumeric characters and underscores. The username may not "
+        + "start or end with any underscores."
+        + "3. The username must not have consecutive underscores.";
+
+    /*
+     * The first character of the telegram must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "^@(?=.{5,32}$)(?!.*__)[A-Za-z][A-Za-z0-9_]*[A-Za-z0-9]$";
+    public static final String TELEGRAM_LINK_PREFIX = "https://t.me/";
+
+    private static final String CONTACT_TYPE_NAME = "Telegram";
+
+    /**
+     * Constructs an {@code Address}.
+     *
+     * @param username A valid username.
+     */
+    public Telegram(String username) {
+        super(CONTACT_TYPE_NAME, TELEGRAM_LINK_PREFIX + username);
+        requireNonNull(username);
+        checkArgument(isValidTelegram(username), MESSAGE_CONSTRAINTS);
+        setValue(username);
+    }
+
+    /**
+     * Returns true if a given string is a valid telegram username.
+     */
+    public static boolean isValidTelegram(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * @return
+     */
+    @Override
+    public String getLink() {
+        return TELEGRAM_LINK_PREFIX + this.getValue().substring(1);
+    }
+
+    @Override
+    public String toString() {
+        return getValue();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof Telegram // instanceof handles nulls
+            && getValue().equals(((Telegram) other).getValue())); // state check
+    }
+
+    @Override
+    public int hashCode() {
+        return getValue().hashCode();
+    }
+}
```

Comment:
We can put these Logic to `Contact` class so we don't need to repeat every time.